packages/core-modules/test/contracts/modules/ElectionModule.vote.test.js

```diff
@@ -0,0 +1,200 @@
+        await fastForwardTo(nominationPeriodStartDate, ethers.provider);
+      });
+
+      it('shows that the current period is Nomination', async function () {
+        assertBn.eq(await ElectionModule.getCurrentPeriodType(), ElectionPeriod.Nomination);
+      });
+
+      describe('when nominations exist', function () {
+        before('nominate', async function () {
+          await ElectionModule.connect(candidate1).nominate();
+          await ElectionModule.connect(candidate2).nominate();
+          await ElectionModule.connect(candidate3).nominate();
+        });
+
+        describe('when entering the vote period', function () {
+          before('fast forward', async function () {
+            await fastForwardTo(votingPeriodStartDate, ethers.provider);
+          });
+
+          it('shows that the current period is Vote', async function () {
+            assertBn.eq(await ElectionModule.getCurrentPeriodType(), ElectionPeriod.Vote);
+          });
+
+          describe('when issuing invalid votes', function () {
+            describe('when voting with zero candidates', function () {
+              it('reverts', async function () {
+                await assertRevert(ElectionModule.elect([]), 'NoCandidates');
+              });
+            });
+
+            describe('when voting candidates that are not nominated', function () {
+              it('reverts', async function () {
+                await assertRevert(
+                  ElectionModule.elect([candidate1.address, candidate2.address, voter1.address]),
+                  'NotNominated'
+                );
+              });
+            });
+
+            describe('when voting duplicate candidates', function () {
+              it('reverts', async function () {
+                await assertRevert(
+                  ElectionModule.elect([
+                    candidate1.address,
+                    candidate2.address,
+                    candidate1.address,
+                  ]),
+                  'DuplicateCandidates'
+                );
+              });
+            });
+          });
+
+          it('can retrieve user vote power', async function () {
+            assertBn.eq(await ElectionModule.getVotePower(voter1.address), 1);
+            assertBn.eq(await ElectionModule.getVotePower(voter2.address), 1);
+            assertBn.eq(await ElectionModule.getVotePower(voter3.address), 1);
+          });
+
+          describe('when issuing valid votes', function () {
+            let ballot1, ballot2, ballot3;
+
+            before('form ballots', async function () {
+              ballot1 = {
+                candidates: [candidate1.address],
+                id: await ElectionModule.calculateBallotId([candidate1.address]),
+              };
+              ballot2 = {
+                candidates: [candidate2.address],
+                id: await ElectionModule.calculateBallotId([candidate2.address]),
+              };
+              ballot3 = {
```

Comment:
it would be nice to have a ballot with more than one candidate.

Reply:
Implemented